From 72466ac801928c205604b99fe01f830809bda930 Mon Sep 17 00:00:00 2001
From: Liquor <lirui130@huawei.com>
Date: Thu, 17 Dec 2020 15:04:56 +0800
Subject: [PATCH] Add check to resolve uname26-version test failed

The uname command is modified in packages uname-build-checks,
but the uname26-version test case needs to use the uname -r query
result.As a result,the test fails.
So we add a judgment to check whether uname-build-checks is installed.
---
 tests/ts/misc/setarch | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/tests/ts/misc/setarch b/tests/ts/misc/setarch
index 7c99cca..25d02c1 100755
--- a/tests/ts/misc/setarch
+++ b/tests/ts/misc/setarch
@@ -77,7 +77,12 @@ ts_finalize_subtest "$finmsg"
 # conditional subtest
 if [ "$uname26_seems_supported" = "yes" ]; then
 ts_init_subtest uname26-version
-	tmp=$($TS_CMD_SETARCH $ARCH --uname-2.6 uname -r)
+	rpm -qa | grep -q "uname-build-checks"
+	if [ $? -eq 0 ]; then
+		tmp=$($TS_CMD_SETARCH $ARCH --uname-2.6 uname.bin -r)
+	else
+		tmp=$($TS_CMD_SETARCH $ARCH --uname-2.6 uname -r)
+	fi
 	if echo "$tmp" | grep -q "^2\.6\."; then
 		echo "kernel version changed to 2.6" >> $TS_OUTPUT
 	else
-- 
2.27.0

